tests/test_uri_templates.py

```diff
@@ -399,3 +434,41 @@ def test_same_level_complex_var(client, reverse):
     assert details_resource.called
     assert details_resource.file_id == file_id_2
     assert details_resource.ext == ext
+
+
+def test_adding_alternate_routes(client):
+    resource_with_alt_routes = ResourceWithAltRoutes()
+    client.app.add_route(
+        '/collections/{collection_id}/items/{item_id}', resource_with_alt_routes)
+    client.app.add_route(
+        '/collections/{collection_id}/items', resource_with_alt_routes, suffix='collection')
+    # GET
+    client.simulate_get('/collections/123/items/456')
+    assert resource_with_alt_routes.collection_id == '123'
+    assert resource_with_alt_routes.item_id == '456'
+    assert resource_with_alt_routes.get_called
+    client.simulate_get('/collections/foo/items')
+    assert resource_with_alt_routes.collection_id == 'foo'
+    # POST
+    client.simulate_post('/collections/foo234/items/foo456')
+    assert resource_with_alt_routes.collection_id == 'foo234'
+    assert resource_with_alt_routes.item_id == 'foo456'
+    assert resource_with_alt_routes.post_called
+    client.simulate_post('/collections/foo123/items')
+    assert resource_with_alt_routes.collection_id == 'foo123'
+    # PUT
+    client.simulate_put('/collections/foo345/items/foo567')
+    assert resource_with_alt_routes.collection_id == 'foo345'
+    assert resource_with_alt_routes.item_id == 'foo567'
+    assert resource_with_alt_routes.put_called
+    client.simulate_put('/collections/foo321/items')
+    assert resource_with_alt_routes.collection_id == 'foo321'
+
+
+def test_custom_error_on_alternate_route_not_found(client):
+    resource_with_alt_routes = ResourceWithAltRoutes()
+    try:
+        client.app.add_route(
+            '/collections/{collection_id}/items', resource_with_alt_routes, suffix='bad-alt')
+    except (AltMethodNotFoundError):
```

Comment:
Is there any particular reason surrounding this with parenthesis?

Reply:
No reason. Glancing at the general style just now I see that if it's just one exception then in falcon there isn't parentheses, so I removed those in my last commit.